### PR TITLE
[onert] Rename "controlflow" backend to "builtin"

### PR DESCRIPTION
**Pull request by wateret**
Rename "controlflow" backend to "builtin". As it does much  more than
supporting controlflow operations.

- Provide kernels for special operations
    - Control Flow Operation
    - Permute(Forward tensors between backend boundary) Operation
- Manages Subgraph's In/Output tensors

Related Issue : #4139

ONE-DCO-1.0-Signed-off-by: Hanjoung Lee <hanjoung.lee@samsung.com>

### CODE REVIEW

**Review by hseok-oh — APPROVED** [review]
LGTM

File: runtime/onert/core/include/compiler/BackendManager.h [hunk 1]
```diff
@@ -65,9 +65,9 @@ class BackendManager
 private:
   std::map<std::string, std::unique_ptr<void, dlhandle_destroy_t>> _handle_map;
   std::map<std::string, std::unique_ptr<backend::Backend, backend_destroy_t>> _gen_map;
-  backend::controlflow::Backend *_controlflow{nullptr};
+  backend::builtin::Backend *_controlflow{nullptr};
```
Comment (ragmani) [inline runtime/onert/core/include/compiler/BackendManager.h:68]: How about renaming `_controlflow` too?

File: runtime/onert/core/src/compiler/Compiler.cc [hunk 2]
```diff
@@ -240,7 +240,7 @@ std::shared_ptr<exec::ExecutorMap> Compiler::compile(void)
     for (auto it = contexts.begin(); it != contexts.end(); it++)
     {
       // Controlflow backend is not for actual computaion of operations so it is an exception
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/Compiler.cc:242]: Here too.

File: runtime/onert/core/src/compiler/LoweredGraph.cc [hunk 3]
```diff
@@ -54,7 +54,7 @@ LoweredGraph::LoweredGraph(const ir::Graph &graph, const CompilerOptions &option
   auto &backend_manager = BackendManager::get();
 
   // Always create Controlflow backend context
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/LoweredGraph.cc:56]: here too.

File: runtime/onert/core/src/compiler/LoweredGraph.cc [hunk 4]
```diff
@@ -335,7 +335,7 @@ void LoweredGraph::makeOpSequences(
 void LoweredGraph::manipulateLowerInfo(
   ir::OperandIndexMap<std::unique_ptr<ir::operand::LowerInfo>> &operands_lower_info)
 {
-  const auto controlflow_backend = BackendManager::get().getControlflow();
+  const auto controlflow_backend = BackendManager::get().getBuiltin();
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/LoweredGraph.cc:338]: here too.

File: runtime/onert/core/src/compiler/LoweredGraph.cc [hunk 5]
```diff
@@ -335,7 +335,7 @@ void LoweredGraph::makeOpSequences(
...
 
   // TODO Rather than using NHWC Get frontend layout of this node from IR
   auto factor = ir::operand::PermuteFactor{controlflow_backend, ir::Layout::NHWC};
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/LoweredGraph.cc:341]: here too.

File: runtime/onert/core/src/compiler/TensorRegistries.h [hunk 6]
```diff
@@ -41,10 +41,9 @@ class TensorRegistries
     for (const auto &e : backend_contexts)
     {
       auto tensor_reg = e.second->tensor_registry;
-      if (e.first->config()->id() == backend::controlflow::Config::ID)
+      if (e.first->config()->id() == backend::builtin::Config::ID)
       {
-        _cf_tensor_reg =
-          std::dynamic_pointer_cast<backend::controlflow::TensorRegistry>(tensor_reg);
+        _cf_tensor_reg = std::dynamic_pointer_cast<backend::builtin::TensorRegistry>(tensor_reg);
         if (include_controlflow)
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/TensorRegistries.h:47]: here too.

File: runtime/onert/core/src/compiler/TensorRegistries.h [hunk 7]
```diff
@@ -64,7 +63,7 @@ class TensorRegistries
     return _tensor_regs.cend();
   }
 
-  std::shared_ptr<backend::controlflow::TensorRegistry> getControlflowTensorRegistry() const
+  std::shared_ptr<backend::builtin::TensorRegistry> getControlflowTensorRegistry() const
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/TensorRegistries.h:66]: here too.

File: runtime/onert/core/src/compiler/BackendManager.cc [hunk 8]
```diff
@@ -49,16 +49,16 @@ BackendManager::BackendManager() { loadControlflowBackend(); }
 
 void BackendManager::loadControlflowBackend()
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/BackendManager.cc:50]: here too

**Comment by wateret** [comment]
@ragmani Sorry about not checking those... and thanks! 😄

**Review by ragmani — APPROVED** [review]
LGTM

**Review by hseok-oh — APPROVED** [review]

**Review by chunseoklee — APPROVED** [review]

**Pull request merged**